### PR TITLE
Fixing curriculum issues.

### PR DESCRIPTION
**Pull request by marcelfloe**

### CODE REVIEW

**Comment by marcelfloe** [comment]
closes #403

**Review by Lama64 — DISMISSED** [review]

File: server/src/main/java/de/uftos/services/CurriculumService.java [hunk 1]
```diff
@@ -105,11 +107,31 @@ public CurriculumResponseDto update(String id, CurriculumRequestDto curriculumRe
...
-    return new CurriculumResponseDto(this.repository.save(curriculum));
+            "Could not find a curriculum with this id"));
+
+    List<LessonsCount> newLessonsCounts = newCurriculum.getLessonsCounts().stream().sorted(
+        Comparator.comparing(l -> l.getSubject().getId())).toList();
+
+    List<LessonsCount> oldLessonsCounts = oldCurriculum.getLessonsCounts().stream().sorted(
+        Comparator.comparing(l -> l.getSubject().getId())).toList();
+
+    if (newLessonsCounts.size() != oldLessonsCounts.size()) {
```
[the remark this answers is not in the document]
Comment (marcelfloe) [reply server/src/main/java/de/uftos/services/CurriculumService.java:121]: I guess you mean "when adding a new subject"?
Yeah... I guess that breaks...

[the remark this answers is not in the document]
Comment (marcelfloe) [reply server/src/main/java/de/uftos/services/CurriculumService.java:121]: I guess your suggestion would be the most reasonable approach (probably also for the creation of a new curriculum when creating a new grade)

**Comment by marcelfloe** [comment]
CurriculumTest and GradesTest both create a curriculum, which was automatically created once the grade got created.
The other tests failing probably still have the instances from the other tests, which didn't get deleted.

**Review by SuperMarcomen — APPROVED** [review]
Java part looks fine to me

**Pull request merged**